lib/model/game_state.dart

```diff
@@ -50,10 +51,17 @@ class GameState with ChangeNotifier {
     randomiser.shuffle(deck);
   }
 
+  void viewDiscards() {
+    scrying.addAll(discard);
+    showScryButtons = false;
+    notifyListeners();
+  }
```

Comment:
Thinking it would keep the state cleaner if we didn't reuse the `scrying` variable for this. We could instead introduce a single bool flag, `viewingDiscard`, which we'd use to toggle in the relevant screen. 

In particular, this means we can avoid having `showScryButtons` in game state, which feels slightly polluting because ideally the state shouldn't need to know anything about the UI.